proto/dataplane.proto

```diff
@@ -29,72 +29,101 @@ enum Error {
   ERROR_UNKNOWN_ERROR = 3;
 }
 
-enum IfType {
-  IF_TYPE_UPLINK = 0;
-  IF_TYPE_MANAGEMENT = 1;
-  IF_TYPE_VXLAN = 2;
-}
-
 message GetConfigGenerationRequest {
 }
 
 message GetConfigGenerationResponse {
   uint64 generation = 1;
 }
 
-message Subnet {
-  string cidr = 1;
-  string name = 2;
+
+/* ================================================ */
+/* ==== Interfaces (both underlat and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
 }
 
-message VPC {
-  string id = 1;
-  string name = 2;
-  uint32 vni = 3;
-  repeated Subnet subnets = 4;
+enum IfType {
+  IF_TYPE_ETHERNET = 0;
+  IF_TYPE_VLAN = 1;
+  IF_TYPE_LOOPBACK = 2;
+  IF_TYPE_VTEP = 3;
 }
 
-message Device {
-  uint32 index = 1;
-  string ipaddr = 2;
-  string name = 3;
-  string pciaddr = 4;
-  IfType type = 5;
+enum IfRole {
+  IF_ROLE_FABRIC = 0;
+  IF_ROLE_MANAGEMENT = 1;
+  IF_ROLE_EXTERNAL = 2;
 }
 
-message PeeringIPs {
+/* ================= */
+/* ==== Overlay ==== */
+/* ================= */
+
+message VpcSubnetAs {
   string cidr = 1;
-  string not = 2;
+  repeated string exclude = 2;
```

Comment:
First of all, it's named `not` in the API, let's please keep it consistent with API. And additionally it shouldn't be list of nots, but we're using list of that objects:

```
ips: # (same for as)
- cidr: 1.2.3.4/25
- not: 1.2.3.5
```

Reply:
Well, the goal of our user-facing API is to be very low-level close to the implementation so I think it's very important to stay consistent as we just agreed on it a few weeks ago - incl. structure, naming and etc. I'm good with changing the user-facing API, but I think creating internal one that will be just slightly different would cause many issues in the future.

Just extending an example, our current API is:

```
ips: # (same for as)
- cidr: 1.0.0.0/8
- cidr: 2.0.0.0/24
- not: 1.2.3.5/25
- not: 2.5.6.7/27
```

So, it's a list of either CIDRs or "not" CIDRs

Reply:
Ok, but that means that the example that I made is not possible with the current API ?

Also, and most importantly, I think we've incurred some mis-alignment when you say that the user-facing API "is very low level", because we've been pursuing the opposite in the last week.

Reply:
This example was possible with the Matov's version IIRC. I wasn't involved into any API discussions last week.